playback/tape_cassettes/s3/s3_tape_cassette.py

```diff
@@ -200,7 +201,14 @@ def iter_recording_ids(self, category, start_date=None, end_date=None, metadata=
 
             def content_filter(recording_str):
                 recording_metadata = decode(recording_str)
-                return all(recording_metadata.get(k) == v for k, v in metadata.items())
+                for k, v in metadata.items():
```

Comment:
Doesn't this mean all metadata values need to support the filter and not just one? This is a bit problematic.